plugins/module_utils/redfish_utils.py

```diff
@@ -1147,6 +1170,26 @@ def manage_power(self, command, resource_uri, action_name):
         response = self.post_request(self.root_uri + action_uri, payload)
         if response['ret'] is False:
             return response
+
+        # If requested to wait for the service to be available again, block
+        # until it's ready
+        if wait:
+            start_time = time.time()
+            # Start with a large enough sleep.  Some services will process new
+            # requests while in the middle of shutting down, thus breaking out
+            # early.
+            time.sleep(30)
+
+            # Periodically check for the service's availability.
+            # Even if the timeout is exhausted, we still want to return success
+            # since the power/reset operation was successful.
+            while wait_timeout > 0:
+                status = self.check_service_availability()
+                if status['available']:
+                    # It's available; we're done
+                    break
```

Comment:
@glimchb if we never get to this `break`, should this be an error condition? Essentially this happens if we exhaust our wait time. I can see arguments for/against either way... I'm going to ask others in DMTF about their preference, but I'd like your opinion as well.

Reply:
I see in other places similar wait failed, for example:
https://github.com/ansible-collections/community.aws/blob/stable-7/plugins/modules/mq_broker.py#L403-L440

if we are not failing, should at least add return value if we reached timeout or availability was success...

Reply:
Sounds good to me; I like showing an error that the service didn't return. That'll help prevent users from simply continuing on with their workflow, which will likely start to fail right after this step. I'll see if others agree.

Reply:
Done